### ISSUE TITLE
How the KM curve is drawn？

### ISSUE THREAD

**Issue by hljWmh** [ajv012/SurvPath#15]
I didn't quite understand, is this KM curve drawn with validation set data or training set? Is there any specific code for this, as the article doesn't explain this in detail. After the 5-fold cross-training, is the one with the best results chosen to draw the KM curve?

**Comment by ajv012** [comment]
We do not provide the code for drawing KM curves as it is pretty standard code from [lifelines](https://lifelines.readthedocs.io/en/latest/fitters/univariate/KaplanMeierFitter.html). However, here's an overview of the process. We first use the model from each fold post training to predict the risk scores on the validation set for that fold. Then we concat the risk scores for all of the validation folds, giving us the risk scores for the entire dataset. We then use lifelines to fit a KM fitter and plot in 2 risk groups (high and low). The same package is used to perform the logrank statistical test between risk groups as well. 

Please let us know if something is unclear and we would be happy to help!

**Comment by hljWmh** [comment]
Is the distinction between high and low risk groups based on the median? Also, I still have a question if it is feasible to construct a risk score for the overall dataset using the risk scores obtained from the validation set for each fold? Since the training data is different for each fold, the weights obtained should be different, right? Won't this direct integration cause some irregularities?
Thank you for your patience!


王默涵
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mahmoodlab/SurvPath"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年12月17日(星期二) 凌晨5:57
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mahmoodlab/SurvPath] How the KM curve is drawn？ (Issue #15)





 
We do not provide the code for drawing KM curves as it is pretty standard code from lifelines. However, here's an overview of the process. We first use the model from each fold post training to predict the risk scores on the validation set for that fold. Then we concat the risk scores for all of the validation folds, giving us the risk scores for the entire dataset. We then use lifelines to fit a KM fitter and plot in 2 risk groups (high and low). The same package is used to perform the logrank statistical test between risk groups as well.
 
Please let us know if something is unclear and we would be happy to help!

**Comment by XiaoXueShengwangrui** [comment]
> I didn't quite understand, is this KM curve drawn with validation set data or training set? Is there any specific code for this, as the article doesn't explain this in detail. After the 5-fold cross-training, is the one with the best results chosen to draw the KM curve?

您好，请问您解决了绘制KM曲线的问题了吗？能否分享一下代码呀~

**Comment by sjjadsa** [comment]
@XiaoXueShengwangrui 请问您解决了绘制km曲线的问题了吗？能否给我些指导
